src/types/iterator.rs

```diff
@@ -38,7 +38,7 @@ impl<'p> PyIterator<'p> {
             // Returns NULL if an object cannot be iterated.
             if ptr.is_null() {
                 PyErr::fetch(py);
-                return Err(PyDowncastError);
+                return Err(PyDowncastError::new(None, Some("Iterator")));
```

Comment:
Hmm it looks like this is the only situation where you're not able to provide a source input. Reading through this function I actually think we can simplify it a lot, so I'm going to open a PR for it right now so that you don't need `PyDowncastError` at all here.